sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/IcebergIOLT.java

```diff
@@ -0,0 +1,213 @@
+
+  @BeforeClass
+  public static void beforeClass() {
+    PipelineOptionsFactory.register(IcebergLoadTestPipelineOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(IcebergLoadTestPipelineOptions.class);
+
+    testConfiguration = TEST_CONFIGS.get(options.getTestSize());
+
+    org.apache.beam.sdk.schemas.Schema.Builder builder =
+        org.apache.beam.sdk.schemas.Schema.builder();
+    for (int i = 0; i < testConfiguration.numFields(); i++) {
+      builder = builder.addByteArrayField("bytes_" + i);
+    }
+    beamSchema = builder.build();
+    icebergSchema = SchemaAndRowConversions.beamSchemaToIcebergSchema(beamSchema);
+  }
+
+  @Before
+  public void setUp() {
+    warehouseLocation =
+        String.format(
+            "%s/IcebergIOLT/%s/%s",
+            options.getTempLocation(), testName.getMethodName(), UUID.randomUUID());
+
+    tableId =
+        TableIdentifier.of(
+            testName.getMethodName(), "table" + Long.toString(UUID.randomUUID().hashCode(), 16));
+  }
+
+  @Test
+  public void testReadWriteHadoopType() {
+    Configuration catalogHadoopConf = new Configuration();
+    catalogHadoopConf.set("fs.gs.project.id", options.getProject());
+    catalogHadoopConf.set("fs.gs.auth.type", "SERVICE_ACCOUNT_JSON_KEYFILE");
+    catalogHadoopConf.set(
+        "fs.gs.auth.service.account.json.keyfile", System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+    Catalog catalog = new HadoopCatalog(catalogHadoopConf, warehouseLocation);
+    catalog.createTable(tableId, icebergSchema);
+
+    Map<String, Object> config =
+        ImmutableMap.<String, Object>builder()
+            .put("table", tableId.toString())
+            .put(
+                "catalog_config",
+                ImmutableMap.<String, String>builder()
+                    .put("catalog_name", "hadoop")
+                    .put("catalog_type", CatalogUtil.ICEBERG_CATALOG_TYPE_HADOOP)
+                    .put("warehouse_location", warehouseLocation)
+                    .build())
+            .build();
+
+    PCollection<Long> source =
+        writePipeline.apply(GenerateSequence.from(0).to(testConfiguration.numRows()));
+
+    PCollection<Row> inputRows =
+        source
+            .apply(
+                "Create Rows",
+                MapElements.into(TypeDescriptors.rows())
+                    .via(
+                        new GenerateRow(
+                            testConfiguration.numFields(),
+                            testConfiguration.byteSizePerField(),
+                            beamSchema)))
+            .setRowSchema(beamSchema);
+
+    PCollectionRowTuple.of("input", inputRows)
+        .apply(Managed.write(Managed.ICEBERG).withConfig(config));
+    writePipeline.run().waitUntilFinish();
+
+    // read pipeline
+    PCollection<Long> countRows =
+        PCollectionRowTuple.empty(readPipeline)
+            .apply(Managed.read(Managed.ICEBERG).withConfig(config))
+            .get("output")
```

Comment:
Noting that we should continue to work on ways to make this simpler to use (not having to build a PCollectionTuple for input / being able to grab standard parts of the output without using magic strings)

Reply:
Done in a separate PR (#31470). Updating PR to use the simplified way